Broken build on NixOS

Running NixOS with stable Nixpkgs 25.05 (flakes). During the package build, the compilation fails because `src/main.cpp` uses `addDispatcher` (lines 515-517), but that function is deprecated. Right now I cannot update Hyprland because of this. Error message:

```shell
Hyprspace> ninja: build stopped: subcommand failed.
error: builder for '/nix/store/gnwvq71adivc8l8abq68bmbc7cbadvn7-Hyprspace-0.1+date=2025-05-18_511d399.drv' failed with exit code 1;
       last 25 log lines:
       >       |                                  ^~~~~~~~~~~~
       > [4/6] Compiling C++ object libHyprspace.so.p/src_Input.cpp.o
       > [5/6] Compiling C++ object libHyprspace.so.p/src_main.cpp.o
       > ../src/main.cpp: In function 'PLUGIN_DESCRIPTION_INFO pluginInit(void*)':
       > ../src/main.cpp:513:31: warning: 'bool HyprlandAPI::addDispatcher(void*, const std::string&, std::function<void(std::__cxx11::basic_string<char>)>)' is deprecated [-Wdeprecated-declarations]
       >   513 |     HyprlandAPI::addDispatcher(pHandle, "overview:toggle", ::dispatchToggleOverview);
       >       |     ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > In file included from /nix/store/ls6rxhiwashf0v1qw6nqxkgg4gr61z5k-hyprland-0.50.0+date=2025-07-23_31cc7f3-dev/include/hyprland/src/plugins/PluginSystem.hpp:5,
       >                  from ../src/main.cpp:1:
       > /nix/store/ls6rxhiwashf0v1qw6nqxkgg4gr61z5k-hyprland-0.50.0+date=2025-07-23_31cc7f3-dev/include/hyprland/src/plugins/PluginAPI.hpp:249:33: note: declared here
       >   249 |     APICALL [[deprecated]] bool addDispatcher(HANDLE handle, const std::string& name, std::function<void(std::string)> handler);
       >       |                                 ^~~~~~~~~~~~~
       > ../src/main.cpp:514:31: warning: 'bool HyprlandAPI::addDispatcher(void*, const std::string&, std::function<void(std::__cxx11::basic_string<char>)>)' is deprecated [-Wdeprecated-declarations]
       >   514 |     HyprlandAPI::addDispatcher(pHandle, "overview:open", ::dispatchOpenOverview);
       >       |     ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > /nix/store/ls6rxhiwashf0v1qw6nqxkgg4gr61z5k-hyprland-0.50.0+date=2025-07-23_31cc7f3-dev/include/hyprland/src/plugins/PluginAPI.hpp:249:33: note: declared here
       >   249 |     APICALL [[deprecated]] bool addDispatcher(HANDLE handle, const std::string& name, std::function<void(std::string)> handler);
       >       |                                 ^~~~~~~~~~~~~
       > ../src/main.cpp:515:31: warning: 'bool HyprlandAPI::addDispatcher(void*, const std::string&, std::function<void(std::__cxx11::basic_string<char>)>)' is deprecated [-Wdeprecated-declarations]
       >   515 |     HyprlandAPI::addDispatcher(pHandle, "overview:close", ::dispatchCloseOverview);
       >       |     ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > /nix/store/ls6rxhiwashf0v1qw6nqxkgg4gr61z5k-hyprland-0.50.0+date=2025-07-23_31cc7f3-dev/include/hyprland/src/plugins/PluginAPI.hpp:249:33: note: declared here
       >   249 |     APICALL [[deprecated]] bool addDispatcher(HANDLE handle, const std::string& name, std::function<void(std::string)> handler);
       >       |                                 ^~~~~~~~~~~~~
       > ninja: build stopped: subcommand failed.
       For full logs, run:
         nix log /nix/store/gnwvq71adivc8l8abq68bmbc7cbadvn7-Hyprspace-0.1+date=2025-05-18_511d399.drv
error: 1 dependencies of derivation '/nix/store/5j2l0sfaad8x72bh5s4g243hyjsj73cc-hm_hyprhyprland.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/s2gxsksfx31zgx1im6lh78l36fh9i3wq-home-manager-files.drv' failed to build
error: 1 dependencies of derivation '/nix/store/0jv5d8c6lwlyy0d6d9g8bcn26jxghm7y-home-manager-generation.drv' failed to build
```

--- a/src/Layout.cpp
+++ b/src/Layout.cpp
@@ -25,9 +25,9 @@ void CHyprspaceWidget::updateLayout() {
     if (active) {
         const auto oActiveWorkspace = pMonitor->m_activeWorkspace;
 
-        for (auto& ws : g_pCompositor->m_workspaces) { // HACK: recalculate other workspaces without reserved area
+        for (auto& ws : g_pCompositor->getWorkspaces()) { // HACK: recalculate other workspaces without reserved area
             if (ws->m_monitor->m_id == ownerID && ws->m_id != oActiveWorkspace->m_id) {
-                pMonitor->m_activeWorkspace = ws;
+                pMonitor->m_activeWorkspace = ws.lock();
                 const auto curRules = std::to_string(pMonitor->activeWorkspaceID()) + ", gapsin:" + PGAPSIN->toString() + ", gapsout:" + PGAPSOUT->toString();
                 if (Config::overrideGaps) g_pConfigManager->handleWorkspaceRules("", curRules);
                 g_pLayoutManager->getCurrentLayout()->recalculateMonitor(ownerID);
@@ -44,7 +44,7 @@ void CHyprspaceWidget::updateLayout() {
 
     }
     else {
-        for (auto& ws : g_pCompositor->m_workspaces) {
+        for (auto& ws : g_pCompositor->getWorkspaces()) {
             if (ws->m_monitor->m_id == ownerID) {
                 const auto curRules = std::to_string(ws->m_id) + ", gapsin:" + PGAPSIN->toString() + ", gapsout:" + PGAPSOUT->toString();
                 if (Config::overrideGaps) g_pConfigManager->handleWorkspaceRules("", curRules);

--- a/src/Overview.cpp
+++ b/src/Overview.cpp
@@ -32,7 +32,7 @@ void CHyprspaceWidget::show() {
 
     if (prevFullscreen.empty()) {
         // unfullscreen all windows
-        for (auto& ws : g_pCompositor->m_workspaces) {
+        for (auto& ws : g_pCompositor->getWorkspaces()) {
             if (ws->m_monitor->m_id == ownerID) {
                 const auto w = ws->getFullscreenWindow();
                 if (w != nullptr && ws->m_fullscreenMode != FSMODE_NONE) {

--- a/src/Render.cpp
+++ b/src/Render.cpp
@@ -216,7 +216,7 @@ void CHyprspaceWidget::draw() {
     // find the lowest and highest workspace id to determine which empty workspaces to insert
     int lowestID = INT_MAX;
     int highestID = 1;
-    for (auto& ws : g_pCompositor->m_workspaces) {
+    for (auto& ws : g_pCompositor->getWorkspaces()) {
         if (!ws) continue;
         // normal workspaces start from 1, special workspaces ends on -2
         if (ws->m_id < 1) continue;

